src/simple-keyring.ts

```diff
@@ -0,0 +1,226 @@
+import {
+  concatSig,
+  decrypt,
+  getEncryptionPublicKey,
+  normalize,
+  personalSign,
+  signTypedData,
+  SignTypedDataVersion,
+} from '@metamask/eth-sig-util';
+import { add0x, Eip1024EncryptedData, Hex, Keyring } from '@metamask/utils';
+import { keccak256 } from 'ethereum-cryptography/keccak';
+import randombytes from 'randombytes';
+
+const _TYPE = 'Simple Key Pair';
+
+export default class SimpleKeyring implements Keyring<string[]> {
+  #wallets: { privateKey: Buffer; publicKey: Buffer }[];
+
+  readonly type: string = _TYPE;
+
+  static type: string = _TYPE;
+
+  constructor(options: string[] = []) {
+    this.#wallets = [];
+
+    // istanbul ignore next
+    this.deserialize(options).catch(() => {
+      throw new Error('Problem deserializing SimpleKeyring');
+    });
+  }
+
+  async serialize() {
+    return this.#wallets.map((a) => a.privateKey.toString('hex'));
+  }
+
+  async deserialize(privateKeys: string[]) {
+    this.#wallets = privateKeys.map((hexPrivateKey) => {
+      const strippedHexPrivateKey = stripHexPrefix(hexPrivateKey);
+      const privateKey = Buffer.from(strippedHexPrivateKey, 'hex');
+      const publicKey = privateToPublic(privateKey);
+      return { privateKey, publicKey };
+    });
+  }
+
+  async addAccounts(numAccounts = 1) {
+    const newWallets = [];
+    for (let i = 0; i < numAccounts; i++) {
+      const privateKey = this.#generateKey();
+      const publicKey = privateToPublic(privateKey);
+      newWallets.push({ privateKey, publicKey });
+    }
+    this.#wallets = this.#wallets.concat(newWallets);
+    const hexWallets = newWallets.map(({ publicKey }) =>
+      add0x(bufferToHex(publicToAddress(publicKey))),
+    );
+    return hexWallets;
+  }
+
+  async getAccounts() {
+    return this.#wallets.map(({ publicKey }) =>
+      add0x(bufferToHex(publicToAddress(publicKey))),
+    );
+  }
+
+  async signTransaction(
+    address: Hex,
+    transaction: TypedTransaction,
+    options: Record<string, unknown> = {},
+  ): Promise<TxData> {
+    const privKey = this.#getPrivateKeyFor(address, options);
+    const signedTx = transaction.sign(privKey);
+    // Newer versions of Ethereumjs-tx are immutable and return a new tx object
+    return signedTx === undefined ? transaction : signedTx;
+  }
+
+  // For eth_sign, we need to sign arbitrary data:
+  async signMessage(
+    address: Hex,
+    data: string,
+    opts = { withAppKeyOrigin: '' },
+  ) {
+    const message = stripHexPrefix(data);
+    const privKey = this.#getPrivateKeyFor(address, opts);
+    const msgSig = ecsign(Buffer.from(message, 'hex'), privKey);
+    const rawMsgSig = concatSig(toBuffer(msgSig.v), msgSig.r, msgSig.s);
+    return rawMsgSig;
+  }
+
+  // For personal_sign, we need to prefix the message:
+  async signPersonalMessage(
+    address: Hex,
+    msgHex: Hex,
+    opts = { withAppKeyOrigin: '' },
+  ) {
+    const privKey = this.#getPrivateKeyFor(address, opts);
+    // const privateKey = Buffer.from(privKey, 'hex');
```

Comment:
This is a change from the original implementation. Why is it no longer necessary to do this?

Reply:
I traced it out, and I actually don't think it was ever necessary.  `privKey` was always a Buffer.  But I guess without TypeScript, the compiler didn't know that.

However, it shouldn't be just sitting here commented out.